### DISCUSSION TITLE
Bachelor Thesis

### DISCUSSION THREAD

**Discussion by Foxsden13** [telekom-security/tpotce#1310] [Q&A]
Hey guys,
first of all I would like to say thank you for that awesome project you are running here.
I am currently writing my bachelor thesis and I am kind of starting to panic a bit because I got stuck trying to set up my own website
using snare's clone function. Unfortunately, time is running out on me so I hope you can help me out here.

I created a wordpress website via docker on a cloud server. Docker is running two containers. One runs wordpress and the other one a mysql database. I also registered a domain that leads to the ip address. So far so good. The site looks great.

My next step is cloning the website.
`#clone --target ##########.net/`

`   
/ Logs will be stored in /opt/snare/clone.log|
------------------------------------>SUMMARY<------------------------------------
        Total number of URLs cloned: 7
        Time elapsed: 0:00:11.290358
        Cloned directory: /opt/snare/pages/###########.net
----------------------------------------------------------------------------------`

So, after that is done, I scp the directory with the files cloner extracted in it onto another server where the tpot is running.

`> scp -P 64295 -r .\########.net\ ########@#######:/home/#########`

From here I change the /opt/tpot/etc/tpot.yml file:

`## Snare Service`
`  snare:`
`    container_name: snare`
`    restart: always`
`    tty: true`
`    command: snare --tanner tanner --debug true --no-dorks true --auto-update false --host-ip 0.0.0.0 --port 80 --page-dir /opt/snare/pages`
`    networks:`
`     - tanner_local`
`    ports:`
`     - "80:80"`
`    image: "dtagdevsec/snare:2204"`
`    depends_on:`
`     - tanner`
`    volumes:`
`     - /home/#######/######.net:/opt/snare/pages`

The last step for me is to change the Dockerfile: 

```
FROM alpine:3.17

Include dist
COPY dist/ /root/dist/


Setup apt
RUN apk -U --no-cache add \
               build-base \
               git \
               linux-headers \

               python3 \
               python3-dev \
	       py3-aiohttp \
	       py3-beautifulsoup4 \
	       py3-gitpython \
	       py3-jinja2 \
	       py3-markupsafe \
	       py3-setuptools \
               py3-pip \
	       py3-pycodestyle \
	       py3-wheel && \


#
# Setup Snare 
    git clone https://github.com/mushorg/snare /opt/snare && \
    cd /opt/snare/ && \
    git checkout 0919a80838eb0823a3b7029b0264628ee0a36211 && \
    cp /root/dist/requirements.txt . && \
    pip3 install --no-cache-dir -r requirements.txt && \
    python3 setup.py install && \
    cd / && \
#    rm -rf /opt/snare && \
    mkdir -p /opt/snare/pages && \
#    clone --target http://example.com && \
    mv /root/dist/pages/* /opt/snare/pages/ && \
#
# Setup configs, user, groups
    addgroup -g 2000 snare && \
    adduser -S -s /bin/ash -u 2000 -D -g 2000 snare && \
    mkdir /var/log/tanner && \
    chown -R snare:snare /opt/snare && \
#   
# Clean up
    apk del --purge \
            build-base \
            linux-headers \
            python3-dev && \
    rm -rf /root/* && \
    rm -rf /tmp/* /var/tmp/* && \
    rm -rf /var/cache/apk/*
#
# Start snare
STOPSIGNAL SIGKILL
USER snare:snare
#CMD snare --tanner tanner --debug true --no-dorks true --auto-update false --host-ip 0.0.0.0 --port 80 --page-dir $(shuf -i 1-10 -n 1)
CMD snare --tanner tanner --debug true --auto-update false --host-ip 0.0.0.0 --port 80 --page-dir /opt/snare/pages/#######.net
```

I commented out the line that says:
`rm -rf /opt/snare && \`
because if I got it right, thats where the directory snare gets deleted. But in my understanding I already put my cloned page in there.

I hope somebody has another idea. I am working on this for nights now trying out many different ideas I am having but without success. 
Best regards and thank you very much!

**Comment by Foxsden13** [comment]
Okay, I got it. I had a wrong path but was able to solve the problem.

**Reply by djoker77** [reply]
Hi @Foxsden13, I know it's been one year ago since you commented, but im in a similar situation like you've been back then, also doing a bachelor thesis on evaluating honeypot products, can you maybe help me with the konfiguration of snare for creating own pages and running the created page in snare, it would be very nice if you could just quickly list the konfiguration you had to do and the exact paths to the files i need to adjust. I cloned the webpage but i I have no clue where to store it or what file i need to open to change the configuration on it. Thanks in advance.
